src/advanced_inv.cpp

```diff
@@ -1892,27 +1997,34 @@ bool advanced_inventory::query_charges( aim_location destarea, const advanced_in
             amount = std::min( weightmax, amount );
         }
     }
+    // handle how many of armour type we can equip (max of 2 per type)
+    if(destarea == AIM_WORN) {
+        const auto &id = sitem.items.front()->typeId();
+        // how many slots are available for the item?
+        const int slots_available = input_amount - g->u.amount_worn(id);
+        // if something is going to the worn pane, there has to be at least 1
+        amount = std::min(slots_available, 1);
```

Comment:
This doesn't work correctly. I have one watch on the ground, one already worn. But when I try to move the one on the ground (using ENTER) to the worn pane, it says the destination is already full. It works if I uses "Move an amount of item" (I can enter `1` and it is moved). Might have something to do with the how `input_amount` is calculated.

Another thing: while not wearing any watch, having two watches on the ground, try to move them both via "Move item stack" shows the input box where I can select the amount. This does not happen when I move them to another place on the ground. It should not happen because the destination (worn pane) can actually hold those 2 items (not currently wearing any), I can move them all. This seems to be caused by `amount = std::min(slots_available, 1);`, which should probably refer to `amount` instead of the `1`. Similar to how it's done in the branch above where the amount is limited by the weight.